app/src/main/java/org/sopt/sample/presentation/signup/view/SignUpActivity.kt

```diff
@@ -1,118 +1,121 @@
-package org.sopt.sample.presentation.signup
+package org.sopt.sample.presentation.signup.view
 
 import android.content.Intent
 import android.os.Bundle
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.widget.addTextChangedListener
-import com.google.android.material.snackbar.Snackbar
+import androidx.databinding.DataBindingUtil
 import org.sopt.sample.R
-import org.sopt.sample.data.remote.AuthService
-import org.sopt.sample.data.remote.RequestSignUpDTO
-import org.sopt.sample.data.remote.ResponseSignUpDTO
-import org.sopt.sample.data.remote.ServicePool
 import org.sopt.sample.databinding.ActivitySignUpBinding
 import org.sopt.sample.presentation.login.view.LoginActivity
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
+import org.sopt.sample.presentation.signup.viewModel.SignUpViewModel
 
 
 class SignUpActivity : AppCompatActivity() {
     lateinit var binding: ActivitySignUpBinding
-    lateinit var loginService: AuthService
+    private val viewModel by lazy { SignUpViewModel() }
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_sign_up)
 
-        binding = ActivitySignUpBinding.inflate(layoutInflater)
-        setContentView(binding.root)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_sign_up)
+        binding.vm = viewModel
+        binding.lifecycleOwner = this
+        binding.etId.setText(viewModel.userIdText.value)
+        binding.etPw.setText(viewModel.userPwText.value)
 
+        binding.etName.addTextChangedListener {
+            activateBtn()
+        }
+
+
+        observeId()
+        observePw()
         activateBtn()
         signUp()
     }
 
-    private fun setTextWatcher() {
+    private fun observeId() {
+        viewModel.isUserIdSuit.observe(this) {
+            if (it == true)
+                binding.layoutEtId.boxStrokeColor = getColor(R.color.blue_500)
+            else
+                binding.layoutEtId.boxStrokeColor = getColor(R.color.red_500)
+            activateBtn()
+        }
+    }
+
+    private fun observePw() {
+        viewModel.isUserPwSuit.observe(this) {
+            if (it == true)
+                binding.layoutEtPw.boxStrokeColor = getColor(R.color.blue_500)
+            else
+                binding.layoutEtPw.boxStrokeColor = getColor(R.color.red_500)
+            activateBtn()
+        }
+    }
 
-        if (binding.idEt.text.length *
-            binding.pwET.text.length *
-            binding.etName.text.length != 0
+    private fun checkAllEditText() {
```

Comment:
이 함수는 btn color를 설정해주는 함수라고 생각이 드네요  
```suggestion
    private fun setButtonColor() {
```

Reply:
맞아요,,,